src/module/sqs/queue-processor.js

```diff
@@ -0,0 +1,55 @@
+const fs = require('smart-fs');
+const path = require('path');
+const Joi = require('joi-strict');
+const { wrap } = require('lambda-async');
+
+module.exports = ({ sendMessageBatch }) => ({ queueUrl, stepsDir }) => {
+  const steps = fs
+    .readdirSync(stepsDir)
+    .reduce((p, step) => Object.assign(p, {
+      [step.slice(0, -3)]: (() => {
+        const { schema, handler, next } = fs.smartRead(path.join(stepsDir, step));
+        return {
+          handler: (payload, event) => {
+            Joi.assert(payload, schema, `Invalid payload received for step: ${payload.name}`);
+            return handler(payload, event);
+          },
+          schema,
+          next
+        };
+      })()
+    }), {});
+
+  return wrap((event) => {
+    if (event.Records.length !== 1) {
+      throw new Error(
+        'Lambda SQS subscription is mis-configured! '
+        + 'Please only process one event at a time for retry resilience.'
+      );
+    }
+    return Promise.all(event.Records.map(async (e) => {
+      const payload = JSON.parse(e.body);
+      if (!(payload instanceof Object && !Array.isArray(payload))) {
+        throw new Error(`Invalid Event Received: ${e.body}`);
```

Comment:
We don't intend to catch these `Errors` here, right? Should this maybe be refactored as
```js
      assert(payload instanceof Object && !Array.isArray(payload), Invalid Event Received: ${e.body})
```